### PR TITLE
[0.9] Backports and CI update

### PR DESCRIPTION
**Pull request by nabobalis**
Backports #2988, #2946  #3077 #3075 #3081

### CODE REVIEW

**Comment by nabobalis** [comment]
Online test fails are real and need to be fixed before merge.

**Comment by nabobalis** [comment]
Doc build seems to be a 404 issue. 
```
  File "/opt/conda/lib/python3.7/urllib/request.py", line 1319, in do_open
    raise URLError(err)
urllib.error.URLError: <urlopen error timed out>
```

File: sunpy/net/vso/vso.py [hunk 1]
```diff
@@ -47,7 +47,9 @@
 TIME_FORMAT = config.get("general", "time_format")
 
 DEFAULT_URL_PORT = [{'url': 'http://docs.virtualsolar.org/WSDL/VSOi_rpc_literal.wsdl',
-                     'port': 'nsoVSOi', 'transport': WellBehavedHttpTransport}]
+                     'port': 'nsoVSOi', 'transport': WellBehavedHttpTransport},
+                    {'url': 'https://sdac.virtualsolar.org/API/VSOi_rpc_literal.wsdl',
```
Comment (Cadair) [inline sunpy/net/vso/vso.py]: Does 0.9 have zeep? I am pretty sure this is a zeep only thing.

Comment (nabobalis) [reply sunpy/net/vso/vso.py]: You mean using that specific wsdl is a zeep feature? 0.9 does not have zeep.

Comment (Cadair) [reply sunpy/net/vso/vso.py]: Yeah I don't think that WDSL will work with suds-jurko because of missing https features.

Comment (nabobalis) [reply sunpy/net/vso/vso.py]: Ok I will undo those commits.

**Comment by Cadair** [comment]
I think one of the online fails here I fixed in #3069

**Pull request merged**